graf2d/fitsio/src/TFITS.cxx

```diff
@@ -509,12 +517,79 @@ Bool_t TFITSHDU::LoadHDU(TString& filepath_filter)
                         for (long component = 0; component < repeat; component++) {
                            vec[component] = array[offset++];
                         }
-                        fCells[cellindex++].fRealVector = vec;
+                        fCells[cellindex++].fRealArray = vec;
                      }
                      delete[] array;
                   }
                }
             }
+         } else if (typecode < 1) {
+            // this column contains variable-length arrays
+            fColumnsInfo[colnum].fType = kRealVector;
+
+            // null variables needed by the fits_read_col
+            double nulval = 0;
+            int anynul=0;
+            
+            // loop over rows to derive the total memory requirement
+            fColumnsInfo[colnum].fRowStart.assign(table_rows+1, 0);
+            fColumnsInfo[colnum].fVarLengths.assign(table_rows, 0);
+            fColumnsInfo[colnum].fRowStart[0] = 0;
+
+            for (long row = 0; row < table_rows; row++) {
+               long offset = 0;
+               long repeat_row = 0; 
+               
+               fits_read_descript(fp, colnum+1, row+1, &repeat_row, &offset, &status);
+               // store the starting of each row and the number of elements it contains 
+               fColumnsInfo[colnum].fRowStart[row+1] = fColumnsInfo[colnum].fRowStart[row] + repeat_row;
+               fColumnsInfo[colnum].fVarLengths[row] = repeat_row;
+            }
+            
+           for (long row = 0; row < table_rows; row++) {
+               // number of elements in the cell we want to read, i.e.
+               // number of elements in the variable-length array
+               int nelements = fColumnsInfo[colnum].fRowStart[row + 1] - fColumnsInfo[colnum].fRowStart[row];
+               // size of the variable-length array
+               const int size = fColumnsInfo[colnum].fVarLengths[row];
+               // vector to store the results
+               TArrayD *vec = new TArrayD(size);
+               // variable-length array have negative DATATYPE
+               int abstype = TMath::Abs(typecode);
+             
+               // define the array to load the data with the CFITSIO functions
+               // a fixed arrays is needed as argument to the fits_read_col function
+               // so a new `DATATYPE array[size]` is defined for each case and then
+               // passed to the `fits_read_col` function
+               //
+               // TODO: add all type cases, for now only short, long, float and double are considered
+               //
+               if (abstype == 21) {
+                  short data[size];
```

Comment:
The version or setup of Microsoft's compiler on Windows, does apparently not support this construct (array which a 'const int' size).  You could try ```std::vector<short> data; data.resize(sizse);```

Reply:
Also done,
I replaced the data with a vector, as per your suggestion and then passed the `vector.data()` array to the `fits_read_col` function requiring it.